pandas/tests/test_window.py

```diff
@@ -2805,20 +2823,25 @@ def expanding_func(x, min_periods=1, center=False, axis=0):
             return getattr(exp, func)()
         self._check_expanding(expanding_func, static_comp, preserve_nan=False)
 
-    def test_expanding_apply(self):
+    def test_expanding_apply(self, raw):
 
         def expanding_mean(x, min_periods=1):
+
             exp = x.expanding(min_periods=min_periods)
-            return exp.apply(lambda x: x.mean())
+            result = exp.apply(lambda x: x.mean(), raw=raw)
+            return result
 
-        self._check_expanding(expanding_mean, np.mean)
+        # TODO(jreback), needed to add preserve_nan=False
```

Comment:
if anyone is interested to see why this doesn't work, would be hepful, @WillAyd (you have some xp in this area)

Reply:
It looks like rows [20:39] of the series in use here contains NA values. `preserve_nan` is assuming that those will still be NA after the function application, but isn't it by design with expanding functions that the NA values do not get preserved?  Ex:

```python
>>> ser = pd.Series(range(5))
>>> ser[2] = np.nan
>>> ser.expanding().sum()
0    0.0
1    1.0
2    1.0
3    4.0
4    8.0
dtype: float64
```

So I think this keyword is necessary by design of the expanding function

Reply:
yeah exactly - so wondering if the current impl (eg master) is actually wrong

Reply:
I haven't looked at the implementation in detail but I'm assuming this ties back to the default argument of `min_periods=1` that the Expanding class uses - I think that would make it almost impossible for NA data to appear while expanding. 

I'd buy the argument that the default here should match the Rolling impl (i.e. 0) and Expanding should deal with "expected" NAs (i.e. forward looking values in the window that just haven't been reached yet) vs "unexpected" NA (data from a previous entry in the window) in  a more robust fashion. 

It would be a breaking change; happy to open the ticket if you agree

Reply:
yeah this might be a bit of an unexplored corner

yeah if i can show a simple example would be great to open an issue

after all expanding is just rolling with the window len of the len of the obj

Reply:
Maybe it shouldn't be though? I.e. should the window size of expanding be `i for i in range(N)` instead of `N for i in range(N)`? The difference can be illustrated looking at the third element below (elements 4 and 5 are expected to differ from window size)

```python
>>> ser = pd.Series([0, np.nan, 2, 3, 4])
In [127]: ser.rolling(window=2, min_periods=2).sum()
Out[127]: 
0    NaN
1    NaN
2    NaN  # Doesn't populate here; there aren't 2 non-NA entries at this point
3    5.0
4    7.0
dtype: float64

In [128]: ser.rolling(window=len(ser), min_periods=2).sum()
Out[128]: 
0    NaN
1    NaN
2    2.0  # Populates here, even though this is the first non-NA record and min_periods is 2
3    5.0
4    9.0
dtype: float64
```